documentation/docs/10-getting-started/40-web-standards.md

```diff
@@ -34,14 +34,20 @@ The [`Headers`](https://developer.mozilla.org/en-US/docs/Web/API/Headers) interf
 import { json } from '@sveltejs/kit';
 
 /** @type {import('./$types').RequestHandler} */
-export function GET(event) {
+export function GET({ request }) {
 	// log all headers
-	console.log(...event.request.headers);
+	console.log(...request.headers);
 
-	return json({
+	// create a Response from JSON
+	const response = json({
 		// retrieve a specific header
-		userAgent: event.request.headers.get('user-agent')
+		userAgent: request.headers.get('user-agent')
 	});
+
+	// set a custom header on the response
+	response.headers.set('x-custom-header', 'potato');
+
```

Comment:
I don't know if this is a good idea actually. I think it's confusing. We use `setHeaders` everywhere else in the docs, so I'm not sure that it's helpful to do it differently here. I think if you want to see how to set headers you should go to that section of the documentation
```suggestion
```

Reply:
I do agree that it's a bit more complex to do it this way, but I was thinking:

- the docs mention being able to get *and* set headers via the available web standard objects
- I agree with [this take](https://github.com/sveltejs/kit/pull/10260#issuecomment-1609925723) that `setHeaders` is not web standards so is kind of against the spirit of the section

Would you prefer to go with `setHeaders` there for consistency?

Reply:
oh also, please see/consider the updated descriptions I put in -- a few more words might have made this sufficiently clear/understandable.

Reply:
Hmm, I didn't see the preceding sentence as GitHub had it collapsed

I agree we should not use `setHeaders` given that this deals with web standards.

But even if we want to avoid `setHeaders`, I still would not write the code as:
```
	// create a JSON Response using a header we received
	const response = json({
		// retrieve a specific header
		userAgent: request.headers.get('user-agent')
	});

	// set a custom header on the Response we will return
	response.headers.set('x-custom-header', 'potato');

	return response;
```

Instead I would write it as:
```
	// create a JSON Response using a header we received
	return json({
		userAgent: request.headers.get('user-agent')
	}, {
		headers: {'x-custom-header': 'potato'}
	});
```

What if we just changed the sentence before from "The Headers interface allows you to read incoming request.headers and set outgoing response.headers:" to "The Headers interface allows you to read incoming request.headers and set outgoing response.headers. For example, you can get the request.headers as shown below:"

Reply:
OK taking this into account [I made some changes](https://github.com/sveltejs/kit/pull/10260/commits/1ccd6eea4b041c8da54e00ab5388daed7775b788) -- please let me know what you think!